src/main/java/Ui.java

```diff
@@ -0,0 +1,185 @@
+                "     Hello! I'm Duke\n" +
+                "     What can I do for you?\n" +
+                "    ____________________________________________________________\n";
+        System.out.println(logo + '\n' + greetMsg);
+    }
+
+    public String getUserInput() {
+        String userInput = sc.nextLine();
+        return userInput;
+    }
+
+    public static void handleBye() {
+        System.out.println("    ____________________________________________________________\n" +
+                "     Bye. Hope to see you again soon!\n" +
+                "    ____________________________________________________________");
+        System.exit(0);
+    }
+
+    public static void handleList(String input, TaskList tasks) throws DukeException {
+        String[] inputStringArr = input.split(" ");
+        try {
+            if (inputStringArr.length > 1) {
+                throw new DukeException("    ____________________________________________________________\n" +
+                        "     ☹ OOPS!!! The list command should not be followed by other text!\n" +
+                        "    ____________________________________________________________\n");
+            } else {
+                System.out.println("    ____________________________________________________________\n" +
+                        "     Here are the tasks in your list:");
+                for (int i = 0; i < tasks.size(); i++) {
+                    int listNum = i + 1;
+                    Task t = tasks.get(i);
+                    if (t.getType().equals("todo")) {
+                        System.out.println("    " + listNum + "." + t.getTypeIcon() + '[' + t.getStatusIcon() + "] " + t.toString());
+                    } else if (t.getType().equals("event")) {
+                        System.out.println("    " + listNum + "." + t.getTypeIcon() + '[' + t.getStatusIcon() + "] " + t.toString() + " (at: " + t.getDate() + ")");
+                    } else {
+                        System.out.println("    " + listNum + "." + t.getTypeIcon() + '[' + t.getStatusIcon() + "] " + t.toString() + " (by: " + t.getDate() + ")");
+                    }
+                }
+                System.out.println("    ____________________________________________________________\n");
+            }
+        } catch(DukeException ex) {
+            System.err.println(ex.getMessage());
+        }
+    }
+
+    public static void handleBadCommand() throws DukeException {
+        try {
+            throw new DukeException("    ____________________________________________________________\n" +
+                    "     ☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n" +
+                    "    ____________________________________________________________\n");
+        } catch(DukeException ex) {
+            System.err.println(ex.getMessage());
+        }
+    }
+
+    public static void handleDone(String input, TaskList tasks, Storage storage) throws DukeException {
```

Comment:
I understand how all these handlers are meant to separate the commands but I feel that the UI class should solely be for user input and that a Parser class can be used to parse these user inputs and transform them into commands that will be executed, exactly like how I have done it in my code. Everything will seem less messy too

Reply:
ah ok cuz i read the task specification which said it deals with interactions with the user so i thought this was the class that would handle printing as well. will re-implement this. thanks!